### DISCUSSION TITLE
How do I set a postgres column to null with JavaScript on a "save row"

### DISCUSSION THREAD

**Discussion by kpturner** [Budibase/budibase#11498] [Help] [answered]
When doing a particular "save row" action on a postgresql table, I need to set one of the columns using JavaScript.  The trouble is, in some cases the column needs to be null.   However, if the JavaScript returns `null` this has the effect of not changing the column at all.  It just stays at its original value in the database. 

@melohagan any ideas?  I am sure I have seen this resolved in some discussion somewhere but I cannot for the life of me find it.

**Comment by melohagan** [comment, marked as answer]
Hey @kpturner 

I think if you save a blank string to a column, then it will be converted to a null in the backend. 

 - Empty string -> null
 - Null -> ignore

**Comment by kpturner** [comment]
I will give it a try - thanks

**Reply by emerson-nascimento** [reply]
I tested here and not work. I'm pass over the same problem. Need to set to null or empty and is not possible.

**Comment by ZimM-LostPolygon** [comment]
@melohagan That doesn't work at all. I also tried it with a Date field, and nothing is even able to change that value - empty string, null, 0, etc...

Is there really no way to set the value to actual null?

**Reply by kpturner** [reply]
My question was based on the premise that I am setting the field with JavaScript. That seemed to work OK.

**Reply by ZimM-LostPolygon** [reply]
Hm, how are you setting it though?.. I was trying to use the "Update Field Value" with a JavaScript-based value, and that doesn't work for me.

**Reply by kpturner** [reply]
On a "Save Row" action you can specify bespoke updates for individual columns.  If you choose JavaScript then the column is set to whatever value the JavaScript function returns. So setting it to blank will make the column `null`
![image](https://github.com/Budibase/budibase/assets/308646/69bf090f-1ac0-4cd5-b1b2-017609e417d3)

**Comment by ecmonsen** [comment]
Is it different for Execute Query actions? If I set a binding to a blank string, it tries to save a blank string to the database column, instead of a null.

**Reply by melohagan** [reply]
I believe this bug issue may be related: https://github.com/Budibase/budibase/issues/11629
